compat/bswap.h

```diff
@@ -74,6 +74,12 @@ static inline uint64_t git_bswap64(uint64_t x)
 }
 #endif
 
+/* available since Clang 12 and GCC 10, i.a. needed for ARM64/aarch64 on Windows */
+#elif defined(__has_builtin) && __has_builtin(__builtin_bswap32) && __has_builtin(__builtin_bswap64)
+
+#define bswap32(x) __builtin_bswap32((x))
+#define bswap64(x) __builtin_bswap64((x))
```

Comment:
I wonder whether we want to make this the first item in the `if` block (before `#if defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__))`). Don't we always want to prefer built-ins if they're available? The original code in this file was written 13 years ago, long before those built-ins were available. What do you think?

Reply:
I agree, this would make sense. This patch could then even be contributed to the Git project individually, as it can stand on its own.

Reply:
Ok, let's update it then. Will update the commit in a few hours from now. Thanks for the review!

Reply:
@dscho Updated. I had to change the logic a bit because the `__has_builtin` macro is not defined on older compiler versions, and was throwing errors in CI because it still uses GCC 8. What do you think of the [updated approach](https://github.com/git-for-windows/git/pull/3916/commits/481c4efa8021f7e117eb1c237015976fff2348a6)?